src/main/java/org/testng/TestNG.java

```diff
@@ -294,7 +295,16 @@ private void parseSuite(String suitePath) {
         s.setThreadCount(this.m_threadCount);
         // If test names were specified, only run these test names
         if (m_testNames != null) {
```

Comment:
The level of nesting is increasing. Can we instead do a 
```java
if (testNames == null) {
    suites.add(suite);
    continue;
}
XmlSuiteUtils.cloneIfContainsTestsWithNamesMatchingAny(suite, testNames);
List<String> missMatchedTestname = XmlSuiteUtils.getMissMatchedTestNames(testNames);
if (CollectionUtils.hasElements(missMatchedTestname)) {
    XmlSuiteUtils.resetField();
    throw new TestNGException("The test(s) <" + Arrays.toString(missMatchedTestname.toArray()) + "> cannot be found.");
}
suites.addAll(XmlSuiteUtils.getCloneSuite());
XmlSuiteUtils.resetField();
```

Reply:
Done.